Darwin/StitchCore/StitchCore/Core/Internal/StitchAppClientImpl.swift

```diff
@@ -58,33 +61,27 @@ internal final class StitchAppClientImpl: StitchAppClient {
                 withDispatchQueue queue: DispatchQueue = DispatchQueue.global()) throws {
         self.dispatcher = OperationDispatcher.init(withDispatchQueue: queue)
         self.routes = StitchAppRoutes.init(clientAppID: clientAppID)
-        self.info = StitchAppClientInfo(clientAppID: clientAppID,
-                                        dataDirectory: config.dataDirectory,
-                                        localAppName: config.localAppName,
-                                        localAppVersion: config.localAppVersion
-        )
-
-        let internalAuth =
-            try StitchAuthImpl.init(
-                requestClient: StitchRequestClientImpl.init(baseURL: config.baseURL,
-                                                            transport: config.transport,
-                                                            defaultRequestTimeout: config.defaultRequestTimeout),
-                authRoutes: self.routes.authRoutes,
-                storage: config.storage,
-                dispatcher: self.dispatcher,
-                appInfo: self.info)
-
-        self._auth = internalAuth
-        self.push = StitchPushImpl.init(
-            requestClient: self._auth,
-            pushRoutes: self.routes.pushRoutes,
-            dispatcher: self.dispatcher
-        )
-        self.coreClient = CoreStitchAppClient.init(authRequestClient: internalAuth, routes: routes)
+        self.info = StitchAppClientInfo(
+            clientAppID: clientAppID,
+            dataDirectory: config.dataDirectory,
+            localAppName: config.localAppName,
+            localAppVersion: config.localAppVersion,
+            networkMonitor: config.networkMonitor,
+            authMonitor: nil)
```

Comment:
No– there's a circular dependency here that's preventing us from doing a proper initialisation. Since our immutable fields haven't been initialised yet, we need to initialise it after the fact.

This is a sign of smelly design. I'm considering pulling this apart and making it cleaner. At the moment, I was simply trying to reach parity with Android. @adamchel , would you rather I pull it apart now, or we do this later?

Reply:
If it's going to take you more than an hour to do it I would say defer for later but it's probably better to get this done now rather than build up tech debt.